src/config/xml.rs

```diff
@@ -0,0 +1,343 @@
+    #[serde(rename = "@context")]
+    pub context: Option<PolicyContext>,
+    #[serde(rename = "@group")]
+    pub group: Option<String>,
+    #[serde(rename = "$value", default)]
+    pub rules: Vec<RuleElement>,
+    #[serde(rename = "@user")]
+    pub user: Option<String>,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq)]
+pub struct RuleAttributes {
+    #[serde(rename = "@max_fds")]
+    pub max_fds: Option<u32>,
+    #[serde(rename = "@min_fds")]
+    pub min_fds: Option<u32>,
+
+    #[serde(rename = "@receive_error")]
+    pub receive_error: Option<String>,
+    #[serde(rename = "@receive_interface")]
+    pub receive_interface: Option<String>,
+    /// deprecated and ignored
+    #[serde(rename = "@receive_member")]
+    pub receive_member: Option<String>,
+    #[serde(rename = "@receive_path")]
+    pub receive_path: Option<String>,
+    #[serde(rename = "@receive_sender")]
+    pub receive_sender: Option<String>,
+    #[serde(rename = "@receive_type")]
+    pub receive_type: Option<MessageType>,
+
+    #[serde(rename = "@send_broadcast")]
+    pub send_broadcast: Option<bool>,
+    #[serde(rename = "@send_destination")]
+    pub send_destination: Option<String>,
+    #[serde(rename = "@send_destination_prefix")]
+    pub send_destination_prefix: Option<String>,
+    #[serde(rename = "@send_error")]
+    pub send_error: Option<String>,
+    #[serde(rename = "@send_interface")]
+    pub send_interface: Option<String>,
+    #[serde(rename = "@send_member")]
+    pub send_member: Option<String>,
+    #[serde(rename = "@send_path")]
+    pub send_path: Option<String>,
+    #[serde(rename = "@send_type")]
+    pub send_type: Option<MessageType>,
+
+    /// deprecated and ignored
+    #[serde(rename = "@receive_requested_reply")]
+    pub receive_requested_reply: Option<bool>,
+    /// deprecated and ignored
+    #[serde(rename = "@send_requested_reply")]
+    pub send_requested_reply: Option<bool>,
+
+    /// deprecated and ignored
+    #[serde(rename = "@eavesdrop")]
+    pub eavesdrop: Option<bool>,
+
+    #[serde(rename = "@own")]
+    pub own: Option<String>,
+    #[serde(rename = "@own_prefix")]
+    pub own_prefix: Option<String>,
+
+    #[serde(rename = "@group")]
+    pub group: Option<String>,
+    #[serde(rename = "@user")]
+    pub user: Option<String>,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum RuleElement {
+    Allow(RuleAttributes),
+    Deny(RuleAttributes),
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct TypeElement {
+    #[serde(rename = "$text")]
+    pub r#type: BusType,
+}
+
+fn resolve_include_path(base_path: impl AsRef<Path>, include_path: impl AsRef<Path>) -> PathBuf {
+    let p = include_path.as_ref();
+    if p.is_absolute() {
+        return p.to_path_buf();
+    }
+
+    error!(
+        "resolve_include_path: {} {}",
+        &base_path.as_ref().display(),
+        &include_path.as_ref().display()
+    );
```

Comment:
As I wrote on the matrix, for all logging messages  (especially warnings and errors):

* let's make them a bit more descriptive.
* don't include function names and internal details like that.
* if something is an error level message, we most likely want to just abort. Otherwise, it's more a warning.